kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/data/StringTable.kt

```diff
@@ -0,0 +1,43 @@
+package io.kotest.data
+
+import io.kotest.assertions.fail
+
+internal data class StringTable(
+   val headers: List<String>,
+   val lines: List<String>,
+   val skipFirstLine: Boolean = false, // for files
+) {
+
+   fun <T> mapRows(fn: (List<String>) -> T): List<T> =
+      rows.map { fn(it.value) }
+
+   val rows: List<IndexedValue<List<String>>> =
+      lines
+         .withIndex()
+         .filterNot { (index, line) ->
+            val skipHeader = index == 0 && skipFirstLine
+            skipHeader || line.startsWith("#") || line.isBlank()
+         }
+         .map { (index, line) -> IndexedValue(index, parseRow(line)) }
+
+   init {
+      rowsShouldHaveSize(headers.size)
+   }
+
+   private fun rowsShouldHaveSize(size: Int) {
+      val invalid = rows
+         .filter { it.value.size != size }
+         .map { it.index }
+      if (invalid.isNotEmpty()) fail("Expected all rows to have size $size, but got rows at lines $invalid")
+   }
+
+   companion object {
+      internal fun parseRow(line: String): List<String> {
+         val notAPipeSeparator = "🫓"
+         return line
+            .replace("\\|", notAPipeSeparator)
+            .split("|")
+            .map { it.trim().replace(notAPipeSeparator, "|") }
+      }
```

Comment:
Could avoid the `notAPipeSeparator` back and forth with a regex 

```suggestion
         return line.split("[^\\]\|").map(String::trim)
      }
```

Reply:
Also I think we should add a note in the docs that whitespace will be trimmed from the cell content 🙂

Reply:
@Kantis that was my first idea, but the issue here is that `hello|world` would be parsed to `row("hell", "world")` 
I had to do a lame imperative for loop in `StringTable.parseRow()` to manage all cases. 
If you have a better solution I am open

Reply:
> @Kantis Could avoid the notAPipeSeparator back and forth with a regex

that was my first idea too but it removes the character before the separator

```kotlin
test("regex? now you have two problems") {
      val input = "hello|world"
      val regex = Regex("(?:[^\\\\])\\|")
      input.split(regex) shouldBe listOf("hello", "world")
}
Element differ at index: [0]
Expected :["hello", "world"]
Actual   :["hell", "world"]
```

I had to do a lame imperative for loop in StringTable.parseRow() to manage all cases.
If you have a better solution I am open

Reply:
found the right regex